src/App.js

```diff
@@ -2,15 +2,28 @@ import "./App.css";
 import React from "react";
 import { Router } from "./frontend/router/Router";
 import { Navbar, SidebarMenu } from "./frontend/components";
-import { useLocation } from "react-router-dom";
+import { matchPath, useLocation } from "react-router-dom";
 import { ToastContainer } from "react-toastify";
 
 const sidebarDisplay = (pathname) => {
+  const isMatch = matchPath(
+    {
+      path: "/post/:postId",
+      exact: true,
+      strict: true,
+    },
+    pathname
+  )
+    ? true
+    : false;
+
   if (
```

Comment:
This works but you can refactor it.
Create an array of those routes you're looking for. 
e.g.  `forbiddenPaths = ["/explore", "/bookmarks", .....]`
and then instead of the OR operators check for the route using `forbiddenPaths.includes(pathname)`

**Also, all this code could be moved to the `Sidebar` component. `App.js` will then look cleaner IMO**